jax/_src/numpy/reductions.py

```diff
@@ -2513,12 +2521,107 @@ def _quantile(a: Array, q: Array, axis: int | tuple[int, ...] | None,
+  if weights is None:
+    a, = promote_dtypes_inexact(a)
+  else:
+    a, q, weights = promote_dtypes_inexact(a, q, weights)
+    #weights = lax.convert_element_type(weights, a.dtype)
+    a_shape = a.shape
+    w_shape = np.shape(weights)
+    if np.ndim(weights) == 0:
+      weights = lax.broadcast_in_dim(weights, a_shape, ())
+      w_shape = a_shape
+    else:
+      w_shape = np.shape(weights)
+    if w_shape != a_shape:
+      if axis is None:
+        raise TypeError("Axis must be specified when shapes of a and weights differ.")
+      if isinstance(axis, tuple):
+        if w_shape != tuple(a_shape[i] for i in axis):
+            raise ValueError("Shape of weights must match the shape of the axes being reduced.")
+        weights = lax.broadcast_in_dim(
+            weights,
+            shape=a_shape,
+            broadcast_dimensions=axis
+        )
+        w_shape = a_shape
+      else:
+        if len(w_shape) != 1 or w_shape[0] != a_shape[axis]:
+            raise ValueError("Length of weights not compatible with specified axis.")
+        weights = lax.expand_dims(weights, (axis,))
+        weights = _broadcast_to(weights, a.shape)
+        w_shape = a_shape
+
+    if squash_nans:
+      nan_mask = ~lax_internal._isnan(a)
+      weights = _where(nan_mask, weights, 0)
+    else:
+      with jax.debug_nans(False):
+        a = _where(any(lax_internal._isnan(a), axis=axis, keepdims=True), np.nan, a)
+
+    total_weight = sum(weights, axis=axis, keepdims=True)
+    a_sorted, weights_sorted = lax_internal.sort_key_val(a, weights, dimension=axis)
+    cum_weights = cumsum(weights_sorted, axis=axis)
+    cum_weights_norm = lax.div(cum_weights, total_weight)
+
+    def _weighted_quantile(qi):
+      qi = lax.convert_element_type(qi, cum_weights_norm.dtype)
+      index_dtype = dtypes.canonicalize_dtype(dtypes.int_)
+      idx = sum(lax.lt(cum_weights_norm, qi), axis=axis, dtype=index_dtype, keepdims=keepdims)
+      idx = lax.clamp(_lax_const(idx, 0), idx, _lax_const(idx, a_sorted.shape[axis] - 1))
+      idx_prev = lax.clamp(idx - 1, _lax_const(idx, 0), _lax_const(idx, a_sorted.shape[axis] - 1))
+      
+      slice_sizes = list(a_shape)
+      slice_sizes[axis] = 1
+      offset_start = q_ndim
+      total_offset_dims = len(a_shape) + q_ndim if keepdims else len(a_shape) + q_ndim - 1
+      dnums = lax.GatherDimensionNumbers(
+        offset_dims=tuple(range(offset_start, total_offset_dims)),
+        collapsed_slice_dims=(axis,),
+        start_index_map=(axis,)
+     )
+      val = lax.gather(a_sorted, idx[..., None], dimension_numbers=dnums, slice_sizes=slice_sizes)
+      val_prev = lax.gather(a_sorted, idx_prev[..., None], dimension_numbers=dnums, slice_sizes=slice_sizes)
+      cw_prev = lax.gather(cum_weights_norm, idx_prev[..., None], dimension_numbers=dnums, slice_sizes=slice_sizes)
+      cw_next = lax.gather(cum_weights_norm, idx[..., None], dimension_numbers=dnums, slice_sizes=slice_sizes)
+      if method == "linear":
```

Comment:
I notice NumPy only supports weighted quantiles for `method="inverted_cdf"`. We should probably do the same, unless there is good reason to be more general here.